packages/flutter/lib/src/material/toggle_buttons.dart

```diff
@@ -1269,11 +1278,12 @@ class _SelectToggleButtonRenderObject extends RenderShiftedBox {
     const double sweepAngle = math.pi / 2.0;
     final RRect rrect = RRect.fromRectAndCorners(
       center,
-      topLeft: borderRadius.topLeft,
-      topRight: borderRadius.topRight,
-      bottomLeft: borderRadius.bottomLeft,
-      bottomRight: borderRadius.bottomRight,
+      topLeft: (borderRadius.topLeft.x * borderRadius.topLeft.y != 0.0) ? borderRadius.topLeft : Radius.zero,
+      topRight: (borderRadius.topRight.x * borderRadius.topRight.y != 0.0) ? borderRadius.topRight : Radius.zero,
+      bottomLeft: (borderRadius.bottomLeft.x * borderRadius.bottomLeft.y != 0.0) ? borderRadius.bottomLeft : Radius.zero,
+      bottomRight: (borderRadius.bottomRight.x * borderRadius.bottomRight.y != 0.0) ? borderRadius.bottomRight : Radius.zero,
```

Comment:
This change fixes the #2 bug.

Reply:
My first instinct was that some kind of elliptical rounded corner to be drawn anyway instead of having none at all. Is this because of the `drawArc` API that we're using? For now, this looks good.

Reply:
Exactly correct.